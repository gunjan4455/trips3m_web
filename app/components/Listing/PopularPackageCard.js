import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Sensor from 'react-visibility-sensor';

import PackageImgTitle from 'components/Listing/PackageImgTitle';
import PackagePriceExperiment from 'components/Listing/PackagePriceExperiment';
import PackageDaysDiscountExperiment from 'components/Listing/PackageDaysDiscountExperiment';
import PackageCities from 'components/Listing/PackageCities';
import PackageIncExc from 'components/Listing/PackageIncExc';
import {
  trackSegment, PACKAGE_CLICKED, PACKAGE_VIEWED, HOTEL_CATEGORY_CLICKED
} from 'actions/segmentEvents';
import { rescue } from 'helpers/utils';
import './PackageCities.scss';
import './Listing.scss';

class PackageCardExperiment extends Component {
  constructor(props) {
    super(props);

    this.state = { hotelId: props.card.id };
    this.sensorVisibility = false;

    this.trackSegmentEvent = this.trackSegmentEvent.bind(this);
    this.renderHotelSelections = this.renderHotelSelections.bind(this);
  }

  onSensorChange(isVisible, ctaTextPackage, toDestination) {
    if (!this.sensorVisibility && isVisible) {
      this.sensorVisibility = isVisible;
      this.trackSegmentEvent(PACKAGE_VIEWED, ctaTextPackage, toDestination);
    }
  };

  handleChange = (hotelId) => {
    this.setState({ hotelId });
    rescue( () => {
      const { card } = this.props;
      const hotelInfo = card.familyPackages[hotelId];
      this.trackSegmentEvent(HOTEL_CATEGORY_CLICKED, `${hotelInfo[0]} Star`, card.destinations || '', hotelInfo[1]);
    });
  };

  trackSegmentEvent(event, ctaTextPackage, toDestination, value) {
    rescue( () => {
      const { card } = this.props;
      const hotelInfo = card.familyPackages[this.state.hotelId];

      trackSegment({
        event: event,
        category: `Destination:${card.toDestination}`,
        section: 'Package List/View Details',
        object: `Package/${card.id}`,
        cta: ctaTextPackage,
        package_id: card.id,
        destination_list: card.toDestination,
        label: `Days:${card.days}::Hotel:${hotelInfo[0]} Star`,
        value: value ? value : hotelInfo[1],
        package_name: card.name
      });
    });
  };

  renderHotelSelections() {
    const { card } = this.props;

    const hotelsIds = Object.keys(card.familyPackages || {});

    if (hotelsIds.length) {
      const hotelCheckboxes = hotelsIds.map(hotelId => {
        return (
          <div key={hotelId} className="iblock hotelStarItem mr8">
            <input
              name={`popular_hotel_star_input_${card.id}`}
              type="radio"
              className="checkbox-common radio-common-round"
              checked={hotelId.toString() === this.state.hotelId.toString()}
              id={`input_${card.id}_${hotelId}`}
              onChange={() => this.handleChange(hotelId)}
              value={hotelId}
            />
            <label className="" htmlFor={`input_${card.id}_${hotelId}`}>
              {card.familyPackages[hotelId][0]} Star
            </label>
          </div>
        );
      });

      return <div className={`row row- star-ratings-selection more-than-three-stars`}>{hotelCheckboxes}</div>;
    }

    return null;
  };

  render() {
    const { card } = this.props;
    // TODO: Ugly code. Handle it differently
    const toDestination = card.destinations;
    const { hotelId } = this.state;
    const ctaTextPackage = 'View Details';
    const { familyPackages = {} } = card;
    const hotelWithPrices = familyPackages && familyPackages[hotelId] ? familyPackages[hotelId] : [];

    const renderChild = () => (
      <div className="relative">
        <PackageImgTitle isLargeImg={true} name={card.destinations} altText={card.imageText}
                         imageUrl={card.imageUrl} imagekitUrl={card.imagekitUrl} />
      </div>
    );

    return (
      <div className="row row- mb8 mt8 relative exp-1 package-card">
        {
          __SERVER__ ?
            <a
              href={`${card.packageUrl}`}
              className="sfcw"
              onClick={() => this.trackSegmentEvent(PACKAGE_CLICKED, ctaTextPackage, toDestination)}
            >
              { renderChild() }
            </a> :
            <Link
              to={`${card.packageUrl}`}
              className="sfcw"
              onClick={() => this.trackSegmentEvent(PACKAGE_CLICKED, ctaTextPackage, toDestination)}
            >
              { renderChild() }
            </Link>
        }
        <Sensor onChange={isVisible => this.onSensorChange(isVisible, ctaTextPackage, toDestination)} intervalDelay={3000}/>
        <div className={`row row- sbcw 'pt8 pb8'}`}>
          <div >
            <div className="row row-">
              <div className="col-xs-12 pl0 pr0 f10">
                <PackageCities cities={card.cityListWithDays || {}}/>
              </div>
            </div>
            <div className="row row-">
              <div className="iblock pl0">
                <PackageDaysDiscountExperiment
                  days={card.days}
                  nights={card.nights}
                  isPackageRevamp={true}
                />
              </div>
              <div className="block">
              { this.renderHotelSelections() }
              </div>
            </div>
            <div className="row row-">
              <div className="col-xs-12 pl0 pr8">
                <PackagePriceExperiment
                  price={hotelWithPrices.length ? hotelWithPrices[2] : card.price}
                  priceUnit={card.miniDesc}
                  currency={card.currency}
                  discount={
                    hotelWithPrices.length
                      ? hotelWithPrices[2] - hotelWithPrices[1]
                      : card.price - card.discountedPrice
                  }
                  promoTag={card.promoTag}
                  totalPrice={card.price}
                  totalDiscount={card.price - card.discountedPrice}
                />
              </div>
            </div>
            <div className="row inclusion-height-box mt8">
              <PackageIncExc inclusions={card.inclusions} id={`card_${card.id}`}
               hotelInfo={card.familyPackages && card.familyPackages[`${hotelId}`]}
               isPackageRevamp={true}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

PackageCardExperiment.propTypes = {
  card: PropTypes.object.isRequired,
};

export default PackageCardExperiment;
