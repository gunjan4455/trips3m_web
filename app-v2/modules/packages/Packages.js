import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

import Swiper from 'components/Swiper/Swiper';
import Heading from '../shared/Heading';
import PopularPackageCard from 'components/Listing/PopularPackageCard';

class Packages extends React.Component {
  render() {
    const { packages, title, heading, travelmonth } = this.props;
    return (
      <div className="clearfix p15 sbcw">
        <div className="flex spaceBetween relative">
          <div className="flexFull">
            <Heading
              data={{
                title,
                heading
              }}
            />
          </div>
          <Link
            to={`/tour-packages/${packages[0].destinations ||
              packages[0].destination}`}
            className="iblock f14 fw7"
          >
            View All
          </Link>
        </div>
        <div className="swiper-full-width">
        <Swiper
          id="packagesSlider"
          spaceBetween={15}
          slidesPerView="auto"
          ssrAnimation={{ width: 280, height: 342 }}
          rebuildOnUpdate
        >
          {packages &&
            packages.length &&
            packages.map((packageData, index) => (
              <div key={index}>
                <PopularPackageCard card={packageData} isPackageRevamp={true}/>
              </div>
            ))}
        </Swiper>
        </div>
      </div>
    );
  }
}

Packages.propTypes = {
  packages: PropTypes.array,
  title: PropTypes.string,
  heading: PropTypes.string,
  travelmonth: PropTypes.number
};

Packages.defaultProps = {
  packages: [],
  title: "Popular Packages",
  heading: 'Unlimited choices. Trusted agents. Best prices. Happy memories.'
};

export default Packages;
