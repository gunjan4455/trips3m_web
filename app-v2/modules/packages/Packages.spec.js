import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import Packages from './Packages';
import PopularPackageCard from 'components/Listing/PopularPackageCard';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  packages: [{
    currency: 'Rupee',
    days: 6,
    nights: 5,
    destinations: 'kerala',
    discount: 3648,
    id: 547,
    imageUrl: 'https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630',
    imagekitUrl: 'https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg',
    inclusions:[],
    mini_desc: 'per person',
    name: 'Amazing Kerala Trip: Hills, Backwaters & Safari',
    price: 20065,
    packageUrl: '/packages/2nights-3days-kerala-munnar-tour?id=597',
    discounted_price:19717 ,
  }]
};

const shallowWrapperPackages = (props) => {
  return shallow(<Packages {...props} />);
};


describe('Package card module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperPackages(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render popularpackagecard', () => {
    const wrapper = shallowWrapperPackages(props);
    expect(wrapper.find(PopularPackageCard)).toHaveLength(1);
  });

});
