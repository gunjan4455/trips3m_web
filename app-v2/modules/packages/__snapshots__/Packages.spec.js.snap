// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package card module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Packages
    heading="Unlimited choices. Trusted agents. Best prices. Happy memories."
    packages={
      Array [
        Object {
          "currency": "Rupee",
          "days": 6,
          "destinations": "kerala",
          "discount": 3648,
          "discounted_price": 19717,
          "id": 547,
          "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
          "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
          "inclusions": Array [],
          "mini_desc": "per person",
          "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
          "nights": 5,
          "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
          "price": 20065,
        },
      ]
    }
    title="Popular Packages"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex spaceBetween relative"
        >
          <div
            className="flexFull"
          >
            <Heading
              data={
                Object {
                  "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                  "title": "Popular Packages",
                }
              }
              heading=""
              isViewAllHidden={false}
              styleHeading={Object {}}
              styleSubHeading={Object {}}
              title=""
              type=""
            />
          </div>
          <Link
            className="iblock f14 fw7"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/tour-packages/kerala"
          >
            View All
          </Link>
        </div>,
        <div
          className="swiper-full-width"
        >
          <Swiper
            id="packagesSlider"
            isFullWidthSwiper={false}
            isSmallSwiper={false}
            loadOnDocumentReady={false}
            overflow="hidden"
            rebuildOnUpdate={true}
            showLoader={false}
            slidesPerView="auto"
            spaceBetween={15}
            ssrAnimation={
              Object {
                "height": 342,
                "width": 280,
              }
            }
          >
            <div>
              <PackageCardExperiment
                card={
                  Object {
                    "currency": "Rupee",
                    "days": 6,
                    "destinations": "kerala",
                    "discount": 3648,
                    "discounted_price": 19717,
                    "id": 547,
                    "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                    "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                    "inclusions": Array [],
                    "mini_desc": "per person",
                    "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                    "nights": 5,
                    "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                    "price": 20065,
                  }
                }
                isPackageRevamp={true}
              />
            </div>
          </Swiper>
        </div>,
      ],
      "className": "clearfix p15 sbcw",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="flexFull"
            >
              <Heading
                data={
                  Object {
                    "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                    "title": "Popular Packages",
                  }
                }
                heading=""
                isViewAllHidden={false}
                styleHeading={Object {}}
                styleSubHeading={Object {}}
                title=""
                type=""
              />
            </div>,
            <Link
              className="iblock f14 fw7"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/tour-packages/kerala"
            >
              View All
            </Link>,
          ],
          "className": "flex spaceBetween relative",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Heading
                data={
                  Object {
                    "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                    "title": "Popular Packages",
                  }
                }
                heading=""
                isViewAllHidden={false}
                styleHeading={Object {}}
                styleSubHeading={Object {}}
                title=""
                type=""
              />,
              "className": "flexFull",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                  "title": "Popular Packages",
                },
                "heading": "",
                "isViewAllHidden": false,
                "styleHeading": Object {},
                "styleSubHeading": Object {},
                "title": "",
                "type": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "View All",
              "className": "iblock f14 fw7",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/tour-packages/kerala",
            },
            "ref": null,
            "rendered": "View All",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Swiper
            id="packagesSlider"
            isFullWidthSwiper={false}
            isSmallSwiper={false}
            loadOnDocumentReady={false}
            overflow="hidden"
            rebuildOnUpdate={true}
            showLoader={false}
            slidesPerView="auto"
            spaceBetween={15}
            ssrAnimation={
              Object {
                "height": 342,
                "width": 280,
              }
            }
          >
            <div>
              <PackageCardExperiment
                card={
                  Object {
                    "currency": "Rupee",
                    "days": 6,
                    "destinations": "kerala",
                    "discount": 3648,
                    "discounted_price": 19717,
                    "id": 547,
                    "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                    "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                    "inclusions": Array [],
                    "mini_desc": "per person",
                    "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                    "nights": 5,
                    "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                    "price": 20065,
                  }
                }
                isPackageRevamp={true}
              />
            </div>
          </Swiper>,
          "className": "swiper-full-width",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div>
                <PackageCardExperiment
                  card={
                    Object {
                      "currency": "Rupee",
                      "days": 6,
                      "destinations": "kerala",
                      "discount": 3648,
                      "discounted_price": 19717,
                      "id": 547,
                      "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                      "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                      "inclusions": Array [],
                      "mini_desc": "per person",
                      "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                      "nights": 5,
                      "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                      "price": 20065,
                    }
                  }
                  isPackageRevamp={true}
                />
              </div>,
            ],
            "id": "packagesSlider",
            "isFullWidthSwiper": false,
            "isSmallSwiper": false,
            "loadOnDocumentReady": false,
            "overflow": "hidden",
            "rebuildOnUpdate": true,
            "showLoader": false,
            "slidesPerView": "auto",
            "spaceBetween": 15,
            "ssrAnimation": Object {
              "height": 342,
              "width": 280,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": <PackageCardExperiment
                  card={
                    Object {
                      "currency": "Rupee",
                      "days": 6,
                      "destinations": "kerala",
                      "discount": 3648,
                      "discounted_price": 19717,
                      "id": 547,
                      "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                      "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                      "inclusions": Array [],
                      "mini_desc": "per person",
                      "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                      "nights": 5,
                      "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                      "price": 20065,
                    }
                  }
                  isPackageRevamp={true}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "card": Object {
                    "currency": "Rupee",
                    "days": 6,
                    "destinations": "kerala",
                    "discount": 3648,
                    "discounted_price": 19717,
                    "id": 547,
                    "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                    "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                    "inclusions": Array [],
                    "mini_desc": "per person",
                    "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                    "nights": 5,
                    "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                    "price": 20065,
                  },
                  "isPackageRevamp": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex spaceBetween relative"
          >
            <div
              className="flexFull"
            >
              <Heading
                data={
                  Object {
                    "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                    "title": "Popular Packages",
                  }
                }
                heading=""
                isViewAllHidden={false}
                styleHeading={Object {}}
                styleSubHeading={Object {}}
                title=""
                type=""
              />
            </div>
            <Link
              className="iblock f14 fw7"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/tour-packages/kerala"
            >
              View All
            </Link>
          </div>,
          <div
            className="swiper-full-width"
          >
            <Swiper
              id="packagesSlider"
              isFullWidthSwiper={false}
              isSmallSwiper={false}
              loadOnDocumentReady={false}
              overflow="hidden"
              rebuildOnUpdate={true}
              showLoader={false}
              slidesPerView="auto"
              spaceBetween={15}
              ssrAnimation={
                Object {
                  "height": 342,
                  "width": 280,
                }
              }
            >
              <div>
                <PackageCardExperiment
                  card={
                    Object {
                      "currency": "Rupee",
                      "days": 6,
                      "destinations": "kerala",
                      "discount": 3648,
                      "discounted_price": 19717,
                      "id": 547,
                      "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                      "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                      "inclusions": Array [],
                      "mini_desc": "per person",
                      "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                      "nights": 5,
                      "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                      "price": 20065,
                    }
                  }
                  isPackageRevamp={true}
                />
              </div>
            </Swiper>
          </div>,
        ],
        "className": "clearfix p15 sbcw",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="flexFull"
              >
                <Heading
                  data={
                    Object {
                      "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                      "title": "Popular Packages",
                    }
                  }
                  heading=""
                  isViewAllHidden={false}
                  styleHeading={Object {}}
                  styleSubHeading={Object {}}
                  title=""
                  type=""
                />
              </div>,
              <Link
                className="iblock f14 fw7"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/tour-packages/kerala"
              >
                View All
              </Link>,
            ],
            "className": "flex spaceBetween relative",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Heading
                  data={
                    Object {
                      "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                      "title": "Popular Packages",
                    }
                  }
                  heading=""
                  isViewAllHidden={false}
                  styleHeading={Object {}}
                  styleSubHeading={Object {}}
                  title=""
                  type=""
                />,
                "className": "flexFull",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "data": Object {
                    "heading": "Unlimited choices. Trusted agents. Best prices. Happy memories.",
                    "title": "Popular Packages",
                  },
                  "heading": "",
                  "isViewAllHidden": false,
                  "styleHeading": Object {},
                  "styleSubHeading": Object {},
                  "title": "",
                  "type": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "View All",
                "className": "iblock f14 fw7",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/tour-packages/kerala",
              },
              "ref": null,
              "rendered": "View All",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Swiper
              id="packagesSlider"
              isFullWidthSwiper={false}
              isSmallSwiper={false}
              loadOnDocumentReady={false}
              overflow="hidden"
              rebuildOnUpdate={true}
              showLoader={false}
              slidesPerView="auto"
              spaceBetween={15}
              ssrAnimation={
                Object {
                  "height": 342,
                  "width": 280,
                }
              }
            >
              <div>
                <PackageCardExperiment
                  card={
                    Object {
                      "currency": "Rupee",
                      "days": 6,
                      "destinations": "kerala",
                      "discount": 3648,
                      "discounted_price": 19717,
                      "id": 547,
                      "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                      "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                      "inclusions": Array [],
                      "mini_desc": "per person",
                      "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                      "nights": 5,
                      "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                      "price": 20065,
                    }
                  }
                  isPackageRevamp={true}
                />
              </div>
            </Swiper>,
            "className": "swiper-full-width",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <div>
                  <PackageCardExperiment
                    card={
                      Object {
                        "currency": "Rupee",
                        "days": 6,
                        "destinations": "kerala",
                        "discount": 3648,
                        "discounted_price": 19717,
                        "id": 547,
                        "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                        "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                        "inclusions": Array [],
                        "mini_desc": "per person",
                        "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                        "nights": 5,
                        "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                        "price": 20065,
                      }
                    }
                    isPackageRevamp={true}
                  />
                </div>,
              ],
              "id": "packagesSlider",
              "isFullWidthSwiper": false,
              "isSmallSwiper": false,
              "loadOnDocumentReady": false,
              "overflow": "hidden",
              "rebuildOnUpdate": true,
              "showLoader": false,
              "slidesPerView": "auto",
              "spaceBetween": 15,
              "ssrAnimation": Object {
                "height": 342,
                "width": 280,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <PackageCardExperiment
                    card={
                      Object {
                        "currency": "Rupee",
                        "days": 6,
                        "destinations": "kerala",
                        "discount": 3648,
                        "discounted_price": 19717,
                        "id": 547,
                        "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                        "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                        "inclusions": Array [],
                        "mini_desc": "per person",
                        "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                        "nights": 5,
                        "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                        "price": 20065,
                      }
                    }
                    isPackageRevamp={true}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "card": Object {
                      "currency": "Rupee",
                      "days": 6,
                      "destinations": "kerala",
                      "discount": 3648,
                      "discounted_price": 19717,
                      "id": 547,
                      "imageUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg?tr=,w-630",
                      "imagekitUrl": "https://img.traveltriangle.com/stage/attachments/pictures/842065/original/Thekkady_River.jpg",
                      "inclusions": Array [],
                      "mini_desc": "per person",
                      "name": "Amazing Kerala Trip: Hills, Backwaters & Safari",
                      "nights": 5,
                      "packageUrl": "/packages/2nights-3days-kerala-munnar-tour?id=597",
                      "price": 20065,
                    },
                    "isPackageRevamp": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
